File: Baekjoon/10870_피보나치 수 5/B_10870.js
const fs = require('fs')
const input = fs.readFileSync('/dev/stdin')
const N = parseInt(input)

fib = {
    0: 0,
    1: 1,
    2: 1,
}

function fibonacci(n) {
    if (n == 0) {
        return 0
    } else if (n == 1 || n == 2) {
        return 1
    } else {
        if (fib.hasOwnProperty(n)) {
            return fib[n]
        } else {
            fib[n] = fibonacci(n - 1) + fibonacci(n - 2)
            return fib[n]
        }
    }
}

console.log(fibonacci(N))

File: Baekjoon/10870_피보나치 수 5/B_10870.py
memo = {0: 0, 1: 1, 2: 1}


def fibonacci(a):
    if a in memo:
        return memo[a]
    memo[a] = fibonacci(a - 2) + fibonacci(a - 1)
    return memo[a]


n = int(input())
print(fibonacci(n))
